src/main/java/uk/gov/hmcts/reform/userprofileapi/domain/entities/UserProfile.java

```diff
@@ -117,6 +118,9 @@ public void setStatus(IdamStatus status) {
     }
 
     public void setRoles(IdamRolesInfo idamrolesInfo) {
-        this.roles = idamrolesInfo.getRoles();
+        Optional.ofNullable(idamrolesInfo).ifPresent(
+                idamRolesInfo -> Optional.ofNullable(idamRolesInfo.getRoles())
+                        .ifPresentOrElse(roles -> this.roles = roles,
+                                () -> this.roles = new ArrayList<>()));
```

Comment:
Why do we need to set again () -> this.roles = new ArrayList<>()), because it is already initialised.
List<String> roles = new ArrayList<>();

Reply:
Roles list may contain data and we would want to set it to empty when it is returned as empty from Idam. Without it changes from Idam would be ignored.

Reply:
Great, thanks